PebbleSharp.Core/AppMessage/AppMessagePacket.cs

```diff
@@ -0,0 +1,524 @@
+        //2:        //type
+        //1:0:      //length
+        //1         //value
+
+        //sample outbound message
+        //1:
+        //0:
+        //11:7:71:175:173:135:178:194:147:5:186:182:255:255:255:255:
+        //1:
+        //0:0:0:0:
+        //2:
+        //4:0:
+        //87:195:209:30:
+
+        public byte Command { get; set; }
+
+        public byte TransactionId { get; set; }
+        public UUID ApplicationId { get; set; }
+
+        public AppMessagePacket()
+        {
+            Values = new List<IAppMessageDictionaryEntry>();
+        }
+
+
+
+        public AppMessagePacket(byte[] bytes)
+        {
+			Load(bytes);
+        }
+
+		protected override void Load(byte[] bytes)
+		{
+			Values = new List<IAppMessageDictionaryEntry>();
+			int index = 0;
+			var command = bytes[index];
+			index++;
+
+			Command = command;
+			//if (command == Command)
+			{
+				TransactionId = bytes[index];
+				index++;
+
+				ApplicationId = new UUID(bytes.Skip(index).Take(16).ToArray());
+				index += 16;
+
+				int tupleCount = bytes[index];
+				index++;
+
+				for (int i = 0; i < tupleCount; i++)
+				{
+					uint k;
+					byte t;
+					ushort l;
+
+					k = BitConverter.ToUInt32(bytes, index);
+					index += 4;
+
+					t = bytes[index];
+					index++;
+
+					l = BitConverter.ToUInt16(bytes, index);
+					index += 2;
+
+					IAppMessageDictionaryEntry entry = null;
+					if (t == (byte)PackedType.Bytes)
+					{
+						entry = new AppMessageBytes() { Value = bytes.Skip(index).Take(l).ToArray() };
+					}
+					else if (t == (byte)PackedType.Signed)
+					{
+						if (l == 1)
+						{
+							entry = new AppMessageInt8() { Value = Convert.ToSByte(bytes[index]) };
+						}
+						else if (l == 2)
+						{
+							entry = new AppMessageInt16() { Value = BitConverter.ToInt16(bytes, index) };
+						}
+						else if (l == 4)
+						{
+							entry = new AppMessageInt32() { Value = BitConverter.ToInt32(bytes, index) };
+						}
+						else
+						{
+							throw new InvalidOperationException("Invalid signed integer length");
+						}
+					}
+					else if (t == (byte)PackedType.String)
+					{
+						entry = new AppMessageString() { Value = System.Text.Encoding.UTF8.GetString(bytes, index, l) };
+					}
+					else if (t == (byte)PackedType.Unsigned)
+					{
+						if (l == 1)
+						{
+							entry = new AppMessageUInt8() { Value = bytes[index] };
+						}
+						else if (l == 2)
+						{
+							entry = new AppMessageUInt16() { Value = BitConverter.ToUInt16(bytes, index) };
+						}
+						else if (l == 4)
+						{
+							entry = new AppMessageUInt32() { Value = BitConverter.ToUInt32(bytes, index) };
+						}
+						else
+						{
+							throw new InvalidOperationException("Invalid signed integer length");
+						}
+					}
+					else
+					{
+						throw new InvalidOperationException("Unknown tuple type");
+					}
+					index += l;
+					entry.Key = k;
+					Values.Add(entry);
+				}
+			}
+		}
```

Comment:
There are several instances here where you are throwing an InvalidOperationException. I would suggest that a PebbleException (or a new derived type from it) might be better.